hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -240,12 +252,14 @@ public static void processFiles(FileSystem fs, String basePathStr, Function<File
     }
   }
 
-  public static List<String> getAllPartitionPaths(FileSystem fs, String basePathStr, boolean assumeDatePartitioning)
-      throws IOException {
+  public static List<String> getAllPartitionPaths(FileSystem fs, String basePathStr, boolean useFileListingFromMetadata, boolean verifyListings,
+                                                  boolean assumeDatePartitioning) throws IOException {
     if (assumeDatePartitioning) {
```

Comment:
this should be within another if block:

if(useFileListingFromMetadata) {
}

Reply:
I think there is a fallback code inside `HoodieBackedTableMetadata`. but yeah I ll add one to `HoodieTableMetadata#create()` to return a different impl when `useFileListingFromMetadata=false`

Reply:
I will look at this holistically during https://issues.apache.org/jira/browse/HUDI-1479